File: 9.palindrome-number.go
package leetcode

/*
 * @lc app=leetcode id=9 lang=golang
 *
 * [9] Palindrome Number
 */

// @lc code=start
func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}
	if x < 10 {
		return true
	}
	if x%10 == 0 {
		return false
	}
	var y int
	for x > y {
		y = y*10 + x%10
		x = x / 10
		println(x, y)
	}
	return x == y || x == y/10
}

// @lc code=end

File: 9.palindrome-number.js
/*
 * @lc app=leetcode id=9 lang=javascript
 *
 * [9] Palindrome Number
 */

// @lc code=start
function isPalindrome(x) {
  if (x < 0) {
    return false;
  }

  let num = x;
  let reverse = 0;

  while (num > 0) {
    reverse = reverse * 10 + (num % 10);
    num = Math.floor(num / 10);
  }

  return reverse === x;
}

// @lc code=end
